lib/util.js

```diff
@@ -0,0 +1,62 @@
+module.exports = {
+  logWithPrefix: logWithPrefix,
+  regGetValue: regGetValue,
+  regSearchKeys: regSearchKeys
+}
+
+const log = require('npmlog')
+const execFile = require('child_process').execFile
+const path = require('path')
+
+function logWithPrefix (log, prefix) {
+  function setPrefix(logFunction) {
+    return (...args) => logFunction.apply(null, [prefix, ...args])
```

Comment:
are we going to have complaints about the ES6+ here? this will probably get used with some ancient versions, if this is the first use of arrow functions and spread then maybe we should replace that and hold off.

Reply:
node-gyp requires Node 6 since https://github.com/nodejs/node-gyp/pull/1570. This should not land in v3 (IIUC there won't be more of those anyway), I've added the semver-major label just in case.

This `logWithPrefix` function was moved from `configure.js`, and has been in master for a while (though not in a release).